python/cudf/cudf/core/series.py

```diff
@@ -5418,6 +5418,29 @@ def hash_encode(self, stop, use_name=False):
         mod_vals = hashed_values % stop
         return Series(mod_vals._column, index=self.index, name=self.name)
 
+    @copy_docstring(Frame._interpolate)
+    def interpolate(
+        self,
+        method="linear",
+        axis=0,
+        limit=None,
+        inplace=False,
+        limit_direction=None,
+        limit_area=None,
+        downcast=None,
+        **kwargs,
+    ):
+        return super()._interpolate(
+            method=method,
+            axis=axis,
+            limit=limit,
+            inplace=inplace,
+            limit_direction=limit_direction,
+            limit_area=limit_area,
+            downcast=downcast,
+            **kwargs,
+        )
+
```

Comment:
It's not that I don't want `Index` to share the implementation, it's that the pre/post processing steps for what to do in the Series and DataFrame cases might be different. For instance there's a throw case for DataFrame I included since it seemed quite simple. Rather than having a single `Frame` method that then needs to check to see what its own type actually is and then do something different depending on the result, this seemed cleaner - everything DataFrame specific happens inside `DataFrame`, anything `Series` specific happens in `Series`, and whats common is relegated to `Frame.`

Reply:
I completely agree, that's how it should work. `Frame` should _not_ contain logic checking whether we're operating on a `DataFrame` or a `Series`. My question was more about the fact that in this particular case `Series` doesn't seem to require any special pre- or post-processing at all, so it could just inherit the method without doing any extra work. In general, I think it'll make our lives easier if we put methods like this that need to be exposed by both `DataFrame` and `Series` into the public API (not underscored methods) of `Frame`. In cases where we need to add logic we can always use `super` like you have, but in some cases hopefully we won't.

Reply:
Oh I see what you mean. Yeah, I am +1 on that. Updating.